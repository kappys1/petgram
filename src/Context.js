import React, { createContext, useReducer, useContext } from 'react'
const Context = createContext()

const initialState = {
  isAuth: window.sessionStorage.getItem('token')
}
const reducer = (state, action) => {
  switch (action.type) {
    case 'activeAuth':
      window.sessionStorage.setItem('token', action.token)
      return {
        ...state,
        isAuth: true
      }

    default:
      return state
  }
}

export const Provider = ({ children }) => {
  return (
    <Context.Provider value={useReducer(reducer, initialState)}>
      {children}
    </Context.Provider>
  )
}

export const useStateValue = () => useContext(Context)
