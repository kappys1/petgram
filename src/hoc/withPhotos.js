import { graphql } from 'react-apollo'
import { gql } from 'apollo-boost'

const GET_PHOTOS = gql`
query getPhotos($categoryId: ID){
  photos(categoryId: $categoryId){
    _id
    categoryId
    src
    userId
    liked
    likes
  }
}
`
export const withPhotos = graphql(GET_PHOTOS)
