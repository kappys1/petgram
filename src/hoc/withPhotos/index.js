import { graphql } from '@apollo/client/react/hoc'
import { gql } from 'apollo-boost'

export const GET_PHOTOS = gql`
query getPhotos($categoryId: ID){
  photos(categoryId: $categoryId){
    id
    categoryId
    src
    userId
    liked
    likes
  }
}
`
export const withPhotos = graphql(GET_PHOTOS)
