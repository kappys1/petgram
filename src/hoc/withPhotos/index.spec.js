// Category.test.js
/* global describe, it, expect */
import React from 'react'
import { render } from '@testing-library/react'
import { GET_PHOTOS, withPhotos } from '../withPhotos'
import { Provider } from '../../Context'
import { MockedProvider } from '@apollo/react-testing'

describe('withPhotos', () => {
  it('snapshot', () => {
    const mocks = [
      {
        request: {
          query: GET_PHOTOS
        },
        result: {
          data: {}
        }
      }
    ]
    const { asFragment } = render(
      <Provider>
        <MockedProvider mocks={mocks} addTypename={false}>
          <withPhotos />
        </MockedProvider>
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
