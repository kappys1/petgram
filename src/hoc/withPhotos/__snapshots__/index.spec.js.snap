// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPhotos snapshot 1`] = `
<DocumentFragment>
  <withphotos />
</DocumentFragment>
`;
